packages/autorest.python/autorest/codegen/serializers/model_serializer.py

```diff
@@ -43,14 +43,31 @@ def serialize(self) -> str:
 
     def imports(self) -> FileImport:
         file_import = FileImport()
-        file_import.add_msrest_import(
-            self.code_model, "..", MsrestImportType.Module, TypingSection.REGULAR
-        )
+        if self.code_model.options["models_mode"] == "dpg":
+            file_import.add_submodule_import(
+                "..",
+                "_model_base",
+                ImportType.LOCAL,
+                TypingSection.REGULAR,
+            )
+            file_import.add_submodule_import("typing", "overload", ImportType.STDLIB)
+            file_import.add_submodule_import("typing", "Mapping", ImportType.STDLIB)
+            file_import.add_submodule_import("typing", "Any", ImportType.STDLIB)
+        else:
+            file_import.add_msrest_import(
+                self.code_model, "..", MsrestImportType.Module, TypingSection.REGULAR
+            )
         for model in self.code_model.model_types:
             file_import.merge(model.imports(is_operation_file=False))
-            init_line_parameters = [
-                p for p in model.properties if not p.readonly and not p.is_discriminator
-            ]
+            init_line_parameters = (
+                model.properties
+                if self.code_model.options["models_mode"] == "dpg"
```

Comment:
we don't want readonly or discriminator properties in the init for dpg models either